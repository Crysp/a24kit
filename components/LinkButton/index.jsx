import React from 'react';
import PropTypes from 'prop-types';
import StyledButton from './styled';


/**
 * Иногда ссылку надо выделить. По сути, это кнопка для вызова контекстных действий.
 * @develop
 */
export default class LinkButton extends React.PureComponent {
    static propTypes = {
        /**
         * Текст кнопки
         */
        children: PropTypes.string.isRequired,
        /**
         * Дополнительный CSS класс
         */
        className: PropTypes.string,
    };

    static defaultProps = {
        className: '',
    };

    render() {
        return <StyledButton {...this.props} />;
    }
}
