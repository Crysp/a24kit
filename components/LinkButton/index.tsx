import React from 'react';
import StyledButton from './styled';

export interface Props {
    children: React.ReactNode,
    className: string,
}

/**
 * Иногда ссылку надо выделить. По сути, это кнопка для вызова контекстных действий.
 * @develop
 */
export default class LinkButton extends React.PureComponent<Props> {
    static defaultProps = {
        className: '',
    };

    render() {
        return <StyledButton {...this.props} />;
    }
}
