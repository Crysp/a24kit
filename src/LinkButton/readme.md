```js
<LinkButton>Вызов действия</LinkButton>
```