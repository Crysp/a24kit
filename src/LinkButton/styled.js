import styled from 'styled-components';
import { Color } from '../config';

export default styled.button`
    box-sizing: border-box;
    display: inline-block;
    padding: 12px 24px;
    font: 400 14px "Open Sans", sans-serif;
    line-height: 16px;
    color: ${Color.Blue};
    background-color: ${Color.Zircon};
    border-radius: 2px;
    border: 0;
    text-decoration: none;
    cursor: pointer;
    transition: background-color .2s linear;
    &:hover {
        background-color: #e4edf7;
        text-decoration: underline;
    }
`;
