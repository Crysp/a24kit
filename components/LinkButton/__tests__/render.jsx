import React from 'react';
import renderer from 'react-test-renderer';
import LinkButton from '../index';


describe('Верстка компонента <LinkButton>', () => {
    it('Дефолтная', () => {
        const tree = renderer.create(<LinkButton>Link button</LinkButton>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Дефолтная с дополнительным CSS классом', () => {
        const tree = renderer.create(<LinkButton className="custom">Link button</LinkButton>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
