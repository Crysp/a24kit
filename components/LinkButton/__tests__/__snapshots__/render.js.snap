// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Верстка компонента <LinkButton> Дефолтная 1`] = `
<button
  className="sc-bdVaJa jKnhGB"
>
  Link button
</button>
`;

exports[`Верстка компонента <LinkButton> Дефолтная с дополнительным CSS классом 1`] = `
<button
  className="custom sc-bdVaJa jKnhGB"
>
  Link button
</button>
`;
